### PR TITLE
Fix registering multiple addresses to discovery

### PR DESCRIPTION
**Pull request by wasphin**
See API here at https://github.com/bilibili/discovery/blob/master/doc/api.md#%E6%B3%A8%E5%86%8Cregister

### CODE REVIEW

**Comment by gydong** [comment]
LGTM

**Comment by zyearn** [comment]
Could you add the corresponding Unit Test of this case?

File: src/brpc/policy/discovery_naming_service.cpp [hunk 1]
```diff
@@ -271,9 +272,15 @@ int DiscoveryClient::DoRegister() {
...
-        << "&hostname=" << _params.hostname
-        << "&addrs=" << _params.addrs
-        << "&env=" << _params.env
+        << "&hostname=" << _params.hostname;
+
+    std::vector<butil::StringPiece> addrs;
+    butil::SplitString(_params.addrs, ',', &addrs);
+    for (size_t i = 0; i < addrs.size(); ++i) {
+        os << "&addrs=" << addrs[i];
+    }
```
Comment (zyearn) [inline src/brpc/policy/discovery_naming_service.cpp:281]: According to the comment of butil::SplitString: If several instances of |c| are contiguous, or if |str| begins with or ends with |c|, then an empty string is inserted, we should check if addrs[i] is empty string.

File: test/brpc_naming_service_unittest.cpp [hunk 2]
```diff
@@ -579,6 +579,14 @@ class DiscoveryNamingServiceImpl : public test::DiscoveryNamingService {
                  google::protobuf::Closure* done) {
         brpc::ClosureGuard done_guard(done);
         brpc::Controller* cntl = static_cast<brpc::Controller*>(cntl_base);
+        auto body = cntl->request_attachment().to_string();
+        for (brpc::QuerySplitter sp(body); sp; ++sp) {
+            if (!sp.key().empty()) {
```
Comment (zyearn) [inline test/brpc_naming_service_unittest.cpp:584]: I would suggest this line can be deleted.

Comment (wasphin) [reply test/brpc_naming_service_unittest.cpp]: Thanks for the review, and done.

File: src/brpc/policy/discovery_naming_service.cpp (hunk 1 above)
Comment (wasphin) [reply src/brpc/policy/discovery_naming_service.cpp:283]: Thanks for the review, and done.

**Comment by zyearn** [comment]
Thanks for the contribution.

**Pull request merged**